### PR TITLE
Add CSV review page

### PR DESCRIPTION
**Pull request by ghost**
This PR adds an initial document review UI.

See and example by running `vite`

image

The data handling is an attempt to be quite defensive because I'm not completely sure about the data structure being sent and we also won't get any feedback if this goes wrong (other than via users). Do shout if you spot any gaps in those defenses.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
-  * `build`: build output for importing
-  * `lint`: check and fix linting errors
+- Install [pnpm](https://pnpm.io) globally if you don't have it already `npm i pnpm -g`
+- Install dependencies `pnpm i`
+- Start development server `pnpm dev`
 
-NOTE: When updating this project, remember to run `pnpm build` to publish changes.
+Other useful package scripts:
+
+  * `pnpm test`: run the test suite
```
Comment (DafyddLlyr) [inline README.md:15]: nit: There's no test script in `package.json`

**Review by jessicamcinchak — COMMENTED** [review]
Took a quick first pass through this, it seems like a good direction! 

I will admit on first view, I already want a bit more of that "full fat" data view especially for the CSV (eg metadata, flags, auto-answered yes/no), otherwise I'm worried it feels a bit too much like just wrapping our own version of Kev's CSV->PDF view or the BOPS application view page which also tend to omit some of these things we know are important? Just trying to think about how we make sure partners "get" this feature and its' full potential/value ! 

I'm also curious about still "exposing"/connecting the underlying data formats someway - like a "Download GeoJSON" button under the map, a "Download CSV" button above the list etc. Maybe that's totally redundant because this would always co-exist in a folder alongside those raw data formats, but just thinking out loud here ??

File: src/Answer.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  if (typeof data === "string" || typeof data === "number") {
+    return <span>{data}</span>;
+  }
+  if (Array.isArray(data)) {
+    return List(data);
+  }
+  if (typeof data === "object") {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    return Tree(data);
+  }
```
Comment (DafyddLlyr) [inline src/Answer.tsx:63]: I recognise this logic...! 😉 👍 https://github.com/theopensystemslab/planx-new/pull/1032/files#diff-babdae6181c0ecf3140ce91277e3b570ab98cc7bfe8aee325cc9478c808c1c98R206-R213

File: src/Answer.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    return <Details data={details[0]["value"]} />;
+  }
+  return (
+    <ul>
+      {
+        // eslint-disable-next-line
+        details.map((item) => (
+          <li key={JSON.stringify(item)}>
+            <Details data={item} />
```
Comment (DafyddLlyr) [inline src/Answer.tsx:83]: The syntax here for `data` doesn't match like 75 - is this intentional? 

image

**Review by DafyddLlyr — APPROVED** [review]
This is great - honestly an elegant and nice solution - really looking forward to feedback on this one.

A few nit comments but no showstoppers 👌 

One small suggestion is that it would be helpful to have a Tree and List always visualised by including an object and array in the `example.json`.

File: src/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import "@opensystemslab/map";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace JSX {
+    interface IntrinsicElements {
+      "my-map": MapProps;
+    }
+
+    interface MapProps {
```
Comment (DafyddLlyr) [inline src/Map.tsx:11]: `staticMode` will change this to a "read only" view and remove the zoom buttons.

Docs: https://oslmap.netlify.app/#!/components/vanilla/mymap---basic

File: src/Answer.tsx (hunk 3 above)
Comment (DafyddLlyr) [reply src/Answer.tsx:83]: Aah this makes sense - another example of the downside of not having hard types for flows / nodes / components.

**Pull request merged**